tests/components/metoffice/test_config_flow.py

```diff
@@ -0,0 +1,140 @@
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"api_key": TEST_API_KEY}
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == TEST_SITE_NAME_WAVERTREE
+    assert result2["data"] == {
+        "api_key": TEST_API_KEY,
+        "latitude": TEST_LATITUDE_WAVERTREE,
+        "longitude": TEST_LONGITUDE_WAVERTREE,
+        "name": TEST_SITE_NAME_WAVERTREE,
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_already_configured(hass, requests_mock):
+    """Test we handle duplicate entries."""
+    hass.config.latitude = TEST_LATITUDE_WAVERTREE
+    hass.config.longitude = TEST_LONGITUDE_WAVERTREE
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+
+    all_sites = json.dumps(mock_json["all_sites"])
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.metoffice.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.metoffice.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"api_key": TEST_API_KEY},
+        )
+
+    assert result2["type"] == "create_entry"
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.metoffice.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.metoffice.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"api_key": TEST_API_KEY},
+        )
+
+    assert result2["type"] == "abort"
+    assert result2["reason"] == "already_configured"
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 0
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+async def test_form_cannot_connect(hass, requests_mock):
+    """Test we handle cannot connect error."""
+    hass.config.latitude = TEST_LATITUDE_WAVERTREE
+    hass.config.longitude = TEST_LONGITUDE_WAVERTREE
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text="")
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
```

Comment:
We can pass the user_input as `data` parameter to `async_init`. Then we don't need to call `async_configure`.

Reply:
Gotcha.